### PR TITLE
fix: fix docker login failure when auths username or password is empty which lead to create cluster failure

### PR DESCRIPTION
**Pull request by hellocn9**
fix: fix docker login failure when auths username or password is empty which lead to create cluster failure

### What type of PR is this?
/kind bug

### What this PR does / why we need it:
Fix docker login failure when auths username or password is empty which lead to create cluster failure.
When auths username or password is empty, continue to do next repository's auth login. It means to ignore this repositry's auth.

### Which issue(s) this PR fixes:
Fixes #

### CODE REVIEW

**Comment by 24sama** [comment]
Hi @hellocn9 
Thanks for your contribution!

And I think you can remove this field and make `manifest-list` to use `auth.SkipTLSVerify`:
https://github.com/kubesphere/kubekey/blob/d2703d4b26e4f737b28ed03cf504fa50b613095b/apis/kubekey/v1alpha2/cluster_types.go#L180
https://github.com/kubesphere/kubekey/blob/d2703d4b26e4f737b28ed03cf504fa50b613095b/pkg/images/tasks.go#L335

File: pkg/container/docker.go [hunk 1]
```diff
@@ -83,7 +83,9 @@ func (p *DockerLoginRegistry) Execute(runtime connector.Runtime) error {
 	auths := registry.DockerRegistryAuthEntries(p.KubeConf.Cluster.Registry.Auths)
 
 	for repo, entry := range auths {
-
+		if len(entry.Username) == 0 || len(entry.Password) == 0 {
```
Comment (24sama) [inline pkg/container/docker.go:86]: If all elements of the array are `continue`, then the `docker login` command will not be executed and the `$HOME/.docker/config.json` file will not be generated. And the following comand `mkdir -p /.docker && cp -f $HOME/.docker/config.json /.docker/ && chmod 0644 /.docker/config.json` will report a "Not Found" error.

**Review by 24sama — CHANGES_REQUESTED** [review]

File: pkg/container/docker.go (hunk 1 above)
Comment (hellocn9) [reply pkg/container/docker.go:87]: add judgement for $HOME/.docker/config.json

**Comment by 24sama** [comment]
@hellocn9 
Thanks again!  BTW, please squash your commits, thank you.

**Pull request closed without merging**